nannou/src/window.rs

```diff
@@ -755,6 +755,10 @@ impl<'app> Builder<'app> {
             }
         }
 
+        // Set the class type for X11
```

Comment:
I believe you need to put this behind an `cgf` architecture toggle. The test suite should catch this but for some reason it has not been run...

Reply:
Thanks @Woyten for you review. I considered doing so, but I was unsure which architectures to filter for. Maybe you could help me on this? Which architectures doesn't X run on?
Would it be sufficient to add?:
```
#[cfg(target_family = "unix")]
```

Reply:
It may be that the `cfg` toggle happens in `winit` (https://docs.rs/winit/latest/src/winit/platform/unix.rs.html#252) although filtered on `#[cfg(feature = "x11")]`, which could explain why the test suite doesn't complain(?)

If that is the case, the only thing that gets compiled in is the `use` statement AFAIK.

I will look into where the `feature="x11"` attribute is set. If it is set as a part of the compilation scope of `nannou`, then I would propose using that to toggle the `cfg` setting.

Reply:
> Maybe you could help me on this?

Sure. I expect `if cfg!(unix)` to *not* work since Rust will transform this line to either `if true` or `if false` but it will try to `use winit::platform::unix::WindowBuilderExtUnix` regardless of the architecture. You should probably use
```
#[cfg(target_os = "linux"]
{
  // Your code here
}
```

For an exhaustive list of all target architectures that provide the `WindowBuilderExtUnix` trait have a look at https://docs.rs/winit/0.26.0/src/winit/platform/unix.rs.html#1-7.

Reply:
Thanks for helping out on this. I have a question regarding:
> it will try to use `winit::platform::unix::WindowBuilderExtUnix` regardless of the architecture.

Just so I understand better: So, when using `use` statement inside the `if` block it actually compiles with it? Whereas using `#[cfg(target_os = "linux")]` avoids it becoming compiled into the binary? Just like an `#ifdef` statement in `C++`?

I do get the feeling that using the same `cfg` statement isn't as robust as using the `feature="x11"` attribute. This is however set in `winit`s `Cargo.toml`. Then, how would you get this attribute; how do you get Cargo features form another crate?

Reply:
Ahh, looking at  https://github.com/rust-windowing/winit/blob/master/Cargo.toml#L88, I see that the list used in the `unix.rs` file is the same.

Reply:
> Just so I understand better: So, when using `use` statement inside the `if` block it actually compiles with it? Whereas using `#[cfg(target_os = "linux")]` avoids it becoming compiled into the binary? Just like an `#ifdef` statement in `C++`?

This seems correct!